Update league/oauth2-client dependency to version 2.2.1

Fusion86 opened the pull request:
Currently uses `league/oauth2-client` version 1.3 which in turn uses `ircmaxell/random-lib` version 1.1 which uses the mcrypt extension which is deprecated in PHP 7.1

Updating `league/oauth2-client` to version 2.2.1 solves this issue because it doesn't use `ircmaxell/random-lib` anymore (but a different package which doesn't use mcrypt).

Fusion86 committed:
Update league/oauth2-client dependency to version 2.2.1

shyim commented:
Is here any news?

Fusion86 commented:
This version works fine if you only use it in a 'Login with Discord' kind of way. I haven't tested the other functions but I believe at least some of them are broken.

_Sent from my Motorola XT1092 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

RheaAyase commented:
This PR would break the code, see the changelog for 2.0 version:
```
* Rename getResponse() to getParsedResponse()
* Add getResponse() method that returns the unparsed PSR-7 Response instance
```
And this is the thing I guess: <https://github.com/teamreflex/oauth2-discord/blob/f16528e0d6dd7a652d9154ca1685296e453aa8b6/src/Discord/Discord.php#L127>

Fusion86 committed:
Rename getResponse() to getParsedResponse()

Didn't test it, but should work in theory as far as I can see
Thanks @RheaAyase

Fusion86 commented:
Hmm well that didn't work. It does look like all the errors thrown are completely unrelated though.
Unfortunately I don't use this anymore but fortunately there are alternatives like [RestCord](https://github.com/restcord/restcord), which should work.
